src/flag_gems/ops/glu.py

```diff
@@ -0,0 +1,69 @@
+import torch
+import triton
+import triton.language as tl
+
+@triton.jit
+def glu_kernel(
+    a_ptr, b_ptr, output_ptr,
+    stride_batch, stride_dim,
+    size_batch, size_dim,
+    BLOCK_SIZE: tl.constexpr
+):
+    batch_id = tl.program_id(0)
+    offset = tl.arange(0, BLOCK_SIZE)
+    mask = offset < size_dim
+
+    # 计算偏移：base_ptr + batch_id * stride_batch + offset * stride_dim
+    a_offset = batch_id * stride_batch + offset * stride_dim
+    b_offset = batch_id * stride_batch + offset * stride_dim
+    out_offset = batch_id * stride_batch + offset * stride_dim
+
+    a = tl.load(a_ptr + a_offset, mask=mask)
+    b = tl.load(b_ptr + b_offset, mask=mask)
+
+    b_float = b.to(tl.float32)
+    sigmoid_b = 1 / (1 + tl.exp(-b_float))
+    result = a * sigmoid_b
+
+    tl.store(output_ptr + out_offset, result, mask=mask)
+
+
+def glu(input_tensor: torch.Tensor, dim: int = -1) -> torch.Tensor:
+    
+    dim = dim if dim >= 0 else input_tensor.ndim + dim
+    assert input_tensor.shape[dim] % 2 == 0, "Split dimension must be even"
+
+    # Split into a and b
+    a, b = torch.chunk(input_tensor, 2, dim=dim)
+    output = torch.empty_like(a)
+
+    # Flatten batch and compute strides
+    batch_shape = list(a.shape)
+    size_dim = batch_shape[dim]
+    batch_shape[dim] = 1  # remove GLU dim
+    size_batch = int(torch.prod(torch.tensor(batch_shape)))
+
+    a_contig = a.contiguous()
+    b_contig = b.contiguous()
+    output_contig = output.contiguous()
```

Comment:
consider using utility pointwise_dynamic to deal with non-contiguous tensors. otherwise, torch.contiguous may cause large time cost.